diff --git a/sycl/cmake/modules/FetchEmhash.cmake b/sycl/cmake/modules/FetchEmhash.cmake
index a87f799732d6..e851191fc9c0 100644
--- a/sycl/cmake/modules/FetchEmhash.cmake
+++ b/sycl/cmake/modules/FetchEmhash.cmake
@@ -2,8 +2,17 @@
 if(DEFINED SYCL_EMHASH_DIR OR DEFINED EMHASH_SYS_LOC)
   return()
 endif()
+
 find_file(EMHASH_SYS_LOC "hash_table8.hpp" PATH_SUFFIXES "emhash")
-if(NOT EMHASH_SYS_LOC)
+
+if(EMHASH_SYS_LOC)
+  # If we are using system emhash, include the system directory
+  # containing the emhash folder and note this location so
+  # other modules using emhash can find it.
+  cmake_path(GET EMHASH_SYS_LOC PARENT_PATH EMHASH_SYS_LOC)
+  set(SYCL_EMHASH_DIR ${EMHASH_SYS_LOC} CACHE INTERNAL "")
+  target_include_directories(${LIB_OBJ_NAME} ${EMHASH_SYS_LOC})
+else()
   set(EMHASH_REPO https://github.com/ktprime/emhash)
   message(STATUS "Will fetch emhash from ${EMHASH_REPO}")
   include(FetchContent)
