diff --git a/utils/libclc-remangler/CMakeLists.txt b/utils/libclc-remangler/CMakeLists.txt
index c764eed682f0..ee8e3ffac421 100644
--- a/utils/libclc-remangler/CMakeLists.txt
+++ b/utils/libclc-remangler/CMakeLists.txt
@@ -10,7 +10,7 @@ set(LLVM_LINK_COMPONENTS
   TargetParser
   )
 
-add_clang_tool(libclc-remangler LibclcRemangler.cpp)
+add_llvm_utility(libclc-remangler LibclcRemangler.cpp)
 
 setup_host_tool( libclc-remangler LIBCLC_REMANGLER
   libclc-remangler_exe libclc-remangler_target )
