diff --git a/utils/CMakeLists.txt b/utils/CMakeLists.txt
index 8d8fe15ac682..7f683789a74a 100644
--- a/utils/CMakeLists.txt
+++ b/utils/CMakeLists.txt
@@ -11,7 +11,7 @@ set( LLVM_LINK_COMPONENTS
 )
 
 if( LIBCLC_STANDALONE_BUILD )
-  add_llvm_executable( prepare_builtins prepare-builtins.cpp )
+  add_llvm_utility( prepare_builtins prepare-builtins.cpp )
   set( prepare_builtins_exe prepare_builtins PARENT_SCOPE )
   set( prepare_builtins_target prepare_builtins PARENT_SCOPE )
 else()
