diff --git a/CMakeLists.txt b/CMakeLists.txt
index c3388fee6cc8..e807073e5be0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,11 +1,16 @@
 # Utility project providing various functionalities for SPIR-V devices
 # without native support of these functionalities.
 
+cmake_minimum_required(VERSION 3.20)
+
 set(CMAKE_MODULE_PATH
   ${CMAKE_MODULE_PATH}
   "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules"
   )
 
+find_package(LLVM REQUIRED CONFIG)
+include(AddLLVM)
+
 set(UR_INTREE_SOURCE_DIR "${LLVM_SOURCE_DIR}/../unified-runtime")
 cmake_path(NORMAL_PATH UR_INTREE_SOURCE_DIR OUTPUT_VARIABLE UR_INTREE_SOURCE_DIR)
 
@@ -35,20 +40,23 @@ if(NOT EXISTS ${LIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR})
   get_host_tool_path(sycl-post-link SYCL_POST_LINK sycl-post-link_exe sycl-post-link_target)
   get_host_tool_path(llvm-link LLVM_LINK llvm-link_exe llvm-link_target)
 else()
-  foreach(tool IN ITEMS clang llvm-ar append-file clang-offload-bundler clang-offload-packager file-table-tform llvm-foreach llvm-spirv sycl-post-link llvm-link)
+  foreach(tool IN ITEMS clang llvm-ar append-file clang-offload-bundler clang-offload-packager file-table-tform llvm-foreach llvm-spirv sycl-post-link llvm-link prepare_builtins opt)
     find_program(LLVM_CUSTOM_TOOL_${tool} ${tool}
       PATHS ${LIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR} NO_DEFAULT_PATH)
     set(${tool}_exe ${LLVM_CUSTOM_TOOL_${tool}})
     set(${tool}_target)
+
+    message("Found ${tool} at ${${tool}_exe}")
   endforeach()
 endif()
 
 # Build libdevice for SYCL.
 include(SYCLLibdevice)
 
+include(GNUInstallDirs)
+
 set(LIBDEVICE_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
-set(LIBDEVICE_LIBRARY_DIR ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
-set(LIBDEVICE_RUNTIME_DIR ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+set(LIBDEVICE_LIBRARY_DIR ${CMAKE_INSTALL_LIBDIR})
+set(LIBDEVICE_RUNTIME_DIR ${CMAKE_INSTALL_BINDIR})
 set(LIBDEVICE_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 
-add_subdirectory(test)
