diff --git a/clang/lib/Driver/ToolChains/Clang.cpp b/clang/lib/Driver/ToolChains/Clang.cpp
--- a/clang/lib/Driver/ToolChains/Clang.cpp
+++ b/clang/lib/Driver/ToolChains/Clang.cpp
@@ -10828,9 +10828,8 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
     for (auto &Arg : Cmd->getArguments())
       ForeachArgs.push_back(Arg);

-    SmallString<128> ForeachPath(C.getDriver().Dir);
-    llvm::sys::path::append(ForeachPath, "llvm-foreach");
-    const char *Foreach = C.getArgs().MakeArgString(ForeachPath);
+    const char *Foreach = C.getArgs().MakeArgString(
+        getToolChain().GetProgramPath("llvm-foreach"));
     C.addCommand(std::make_unique<Command>(
         JA, *this, ResponseFileSupport::None(), Foreach, ForeachArgs,
         ArrayRef<InputInfo>{}));
diff --git a/clang/lib/Driver/ToolChains/SYCL.cpp b/clang/lib/Driver/ToolChains/SYCL.cpp
--- a/clang/lib/Driver/ToolChains/SYCL.cpp
+++ b/clang/lib/Driver/ToolChains/SYCL.cpp
@@ -213,9 +213,8 @@ void SYCL::constructLLVMForeachCommand(Compilation &C, const JobAction &JA,
   for (auto &Arg : InputCommand->getArguments())
     ForeachArgs.push_back(Arg);

-  SmallString<128> ForeachPath(C.getDriver().Dir);
-  llvm::sys::path::append(ForeachPath, "llvm-foreach");
-  const char *Foreach = C.getArgs().MakeArgString(ForeachPath);
+  const char *Foreach = C.getArgs().MakeArgString(
+      T->getToolChain().GetProgramPath("llvm-foreach"));

   auto Cmd = std::make_unique<Command>(JA, *T, ResponseFileSupport::None(),
                                        Foreach, ForeachArgs,
@@ -901,9 +901,8 @@ const char *SYCL::Linker::constructLLVMLinkCommand(
       Objs.push_back(
           C.getArgs().MakeArgString(getToolChain().getInputFilename(II)));

   // Get llvm-link path.
-  SmallString<128> ExecPath(C.getDriver().Dir);
-  llvm::sys::path::append(ExecPath, "llvm-link");
-  const char *Exec = C.getArgs().MakeArgString(ExecPath);
+  const char *Exec = C.getArgs().MakeArgString(
+      getToolChain().GetProgramPath("llvm-link"));

   auto AddLinkCommand = [this, &C, &JA, Exec](const char *Output,
                                               const ArgStringList &Inputs,
