diff --git a/cmake/modules/BuildUnifiedRuntime.cmake b/cmake/modules/BuildUnifiedRuntime.cmake
index e90eb7887d14..42f5adffafe2 100644
--- a/cmake/modules/BuildUnifiedRuntime.cmake
+++ b/cmake/modules/BuildUnifiedRuntime.cmake
@@ -69,7 +69,7 @@ if(NOT IS_DIRECTORY "${UR_INTREE_SOURCE_DIR}")
   message(FATAL_ERROR "unified-runtime folder not found at repo root")
 endif()
 
-set(UR_INTREE_BINARY_DIR ${LLVM_BINARY_DIR}/unified-runtime)
+set(UR_INTREE_BINARY_DIR ${CMAKE_BINARY_DIR}/unified-runtime)
 set(UNIFIED_RUNTIME_SOURCE_DIR
   "${UR_INTREE_SOURCE_DIR}" CACHE PATH
   "Path to Unified Runtime Headers" FORCE)
