diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index bb11518b003b..d7c999a4f9f9 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -61,6 +61,7 @@ function(add_sycl_rt_library LIB_NAME LIB_OBJ_NAME)
     PRIVATE
       ${CMAKE_CURRENT_SOURCE_DIR}
       ${sycl_inc_dir}
+      ${SYCL_INCLUDE_BUILD_DIR}
   )
 
   # Object libraries are not linked, so these "libraries" are in fact include
