diff --git a/source/adapters/level_zero/common.cpp b/source/adapters/level_zero/common.cpp
index 0433a2d5..5c2d485f 100644
--- a/source/adapters/level_zero/common.cpp
+++ b/source/adapters/level_zero/common.cpp
@@ -329,7 +329,7 @@ ze_structure_type_t getZeStructureType<ze_device_cache_line_size_ext_t>() {
 template <>
 ze_structure_type_t
 getZeStructureType<ze_intel_device_block_array_exp_properties_t>() {
-  return ZE_INTEL_DEVICE_BLOCK_ARRAY_EXP_PROPERTIES;
+  return static_cast<ze_structure_type_t>(ZE_INTEL_DEVICE_BLOCK_ARRAY_EXP_PROPERTIES);
 }
 #endif // ZE_INTEL_DEVICE_BLOCK_ARRAY_EXP_NAME
 
