diff --git a/source/adapters/level_zero/v2/command_list_cache.cpp b/source/adapters/level_zero/v2/command_list_cache.cpp
index a8dd67fc..94463ef0 100644
--- a/source/adapters/level_zero/v2/command_list_cache.cpp
+++ b/source/adapters/level_zero/v2/command_list_cache.cpp
@@ -16,7 +16,7 @@
 template <>
 ze_structure_type_t
 getZeStructureType<zex_intel_queue_copy_operations_offload_hint_exp_desc_t>() {
-  return ZEX_INTEL_STRUCTURE_TYPE_QUEUE_COPY_OPERATIONS_OFFLOAD_HINT_EXP_PROPERTIES;
+  return static_cast<ze_structure_type_t>(ZEX_INTEL_STRUCTURE_TYPE_QUEUE_COPY_OPERATIONS_OFFLOAD_HINT_EXP_PROPERTIES);
 }
 
 bool v2::immediate_command_list_descriptor_t::operator==(
